txstratum/healthcheck.py

```diff
@@ -0,0 +1,265 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from hathorlib.client import HathorClient
+
+if TYPE_CHECKING:
+    from txstratum.manager import TxMiningManager
+
+
+class ComponentType(str, Enum):
+    """Enum used to store the component types that can be used in the HealthCheckComponentStatus class."""
+
+    DATASTORE = "datastore"
+    INTERNAL = "internal"
+    FULLNODE = "fullnode"
+
+
+class HealthCheckStatus(str, Enum):
+    """Enum used to store the component status that can be used in the HealthCheckComponentStatus class."""
+
+    PASS = "pass"
+    WARN = "warn"
+    FAIL = "fail"
+
+
+@dataclass
+class ComponentHealthCheck:
+    """This class is used to store the result of a health check in a specific component."""
+
+    component_name: str
+    component_type: ComponentType
+    status: HealthCheckStatus
+    output: str
+    time: Optional[str] = None
+    component_id: Optional[str] = None
+    observed_value: Optional[str] = None
+    observed_unit: Optional[str] = None
+
+    def update(self, new_values: Dict[str, Any]) -> None:
+        """
+        Update the object with the new values passed as kwargs.
+
+        Also updates the time field with the current time with the format YYYY-MM-DDTHH:mm:ssZ
+        """
+        self.time = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
```

Comment:
That's a good question, I had to check this.

Apparently this custom time module exists just to facilitate controlling the time during tests.

For the health check logic it doesn't make much different which one we use.

But since they are using it, I'll refactor the health check to use the same, so we keep things consistent.

Reply:
That's a good question, I had to check this.

Apparently they use this custom time module just to be able to easily control time during tests.

For the health check logic this is not needed, and would just add some additional complexity if we try to use their module, since they only wrap the `time.time()` function. I would need the `.gmtime()` and `.strftime()` as well, which I would be able to get only from their `txstratum._time`, which seems ugly.

So I would say we can safely keep the `datetime.utcnow()` in this case as the simplest solution.